src/components/buttonScroll.js

```diff
@@ -0,0 +1,50 @@
+import React, {  useState } from 'react'
+import { Button } from './button'
+
+export function ButtonScroll({ children, ...props }) {
+  const [showScroll, setShowScroll] = useState(false)
+
+  const scrollTop = () =>{
+    window.scrollTo({top: 0, behavior: 'smooth'})
+  };
+
+  const checkScrollTop = () => {
+    if (!showScroll && window.pageYOffset > window.innerHeight){
+      setShowScroll(true)
+    } else if (showScroll && window.pageYOffset <= window.innerHeight){
+      setShowScroll(false)
+    }
+  }
+
+  window.addEventListener('scroll', checkScrollTop)
```

Comment:
É uma boa ideia você envolver este EventListener dentro de um hook de efeito, já que o componente é um componente funcional. Isto faz com que seu componente assim que for desmontado, seu listener também será, aqui vai uma sugestão:
```suggestion
useEffect(() => {
  window.addEventListener('scroll', checkScrollTop);
  return () => window.removeEventListener('scroll', checkScrollTop)
});
```